.github/workflows/integrate.yaml

```
@@ -4,92 +4,121 @@ name: CI
 on:
   workflow_call:
     secrets:
-      charmcraft-credentials:
+      CHARMCRAFT_CREDENTIALS:
+        required: true
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_DEFAULT_REGION:
+        required: true
+      AWS_REGION:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      GH_PERSONAL_ACCESS_TOKEN:
         required: true
 
 jobs:
-
-  lib-check:
-    name: Check libraries
-    runs-on: ubuntu-20.04
-    strategy:
-      matrix:
-        charm:
-        - mlflow-server
+  start-runner:
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-      with:
-        fetch-depth: 0
-    - name: Check libs
-      uses: canonical/charming-actions/check-libraries@2.1.1
-      with:
-        charm-path: ./charms/${{ matrix.charm }}
-        credentials: "${{ secrets.charmcraft-credentials }}"
-        github-token: "${{ secrets.GITHUB_TOKEN }}"
-
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
```

Let's fully pin, and then renovate will manage this

```suggestion
        uses: aws-actions/configure-aws-credentials@v1.7.0
```